### PR TITLE
refactor - Convert whitelist to allowlist

### PR DESCRIPTION
**Pull request by RichardoC**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [X] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [X] **Description:** Provide a concise description of the changes made in this pull request.
- [X] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [X] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
  - <https://github.com/open-webui/docs/pull/202>
- [ ] **Testing:** Have you written and run sufficient tests for validating the changes?
- [X] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [X] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

- Use allowlist to describe allowed models rather than whitelist, as it's becoming the dominant term for this functionality.

### Added

- [List any new features, functionalities, or additions]

### Changed

- Use allowlist rather than whitelist in accordance with <https://inclusivenaming.org/word-lists/tier-1/whitelist/>

### Deprecated

- [List any deprecated functionality or features that have been removed]

---

### Additional Information

- <https://inclusivenaming.org/word-lists/tier-1/whitelist/>

### Screenshots or Videos

- [Attach any relevant screenshots or videos demonstrating the changes]

### CODE REVIEW

**Comment by RichardoC** [comment]
An open question (because I don't know the code well enough) is how to migrate the existing setting for whitelist -> allowlist

File: src/lib/components/admin/Settings/Users.svelte [hunk 1]
```diff
@@ -29,8 +29,8 @@
```
Comment (RichardoC) [inline src/lib/components/admin/Settings/Users.svelte:29]: I'm guessing this line means I don't need to change anything in the storage layer to ensure the whitelist setting is used for the allowlist now.

File: src/lib/i18n/locales/ca-ES/translation.json [hunk 2]
```diff
@@ -408,8 +408,8 @@
 	"Model not selected": "Model no seleccionat",
 	"Model Params": "Paràmetres del model",
 	"Model updated successfully": "Model actualitzat correctament",
-	"Model Whitelisting": "Llista blanca de models",
-	"Model(s) Whitelisted": "Model(s) a la llista blanca",
```
Comment (RichardoC) [inline src/lib/i18n/locales/ca-ES/translation.json:412]: For the other languages, I don't know what the relevant term would be so not changing those in this PR

**Comment by RichardoC** [comment]
Docs PR https://github.com/open-webui/docs/pull/202

File: src/lib/i18n/locales/de-DE/translation.json [hunk 3]
```diff
@@ -408,8 +408,8 @@
 	"Model not selected": "Modell nicht ausgewählt",
 	"Model Params": "Modell-Params",
 	"Model updated successfully": "Modell erfolgreich aktualisiert",
-	"Model Whitelisting": "Modell-Whitelisting",
-	"Model(s) Whitelisted": "Modell(e) auf der Whitelist",
+	"Model Allowlisting": "Modell-Allowlisting",
```
Comment (kaiiiiiiiii) [inline src/lib/i18n/locales/de-DE/translation.json:411]: "Modellauswahl einschränken" (Restrict model selection)

The text should describe the action that i trigger by toggling the toggle button. Should be the case for every other translation as well. Currently it's pretty misleading :)

image

File: src/lib/i18n/locales/de-DE/translation.json [hunk 4]
```diff
@@ -408,8 +408,8 @@
...
+	"Model(s) Allowlisted": "Modell(e) auf der Whitelist",
```
Comment (kaiiiiiiiii) [inline src/lib/i18n/locales/de-DE/translation.json:412]: "Modell(e) zulässig" (Model(s) permitted/allowed)

File: src/lib/i18n/locales/en-GB/translation.json [hunk 5]
```diff
@@ -408,8 +408,8 @@
 	"Model not selected": "",
 	"Model Params": "",
 	"Model updated successfully": "",
-	"Model Whitelisting": "",
-	"Model(s) Whitelisted": "",
+	"Model Allowlisting": "",
```
Comment (kaiiiiiiiii) [inline src/lib/i18n/locales/en-GB/translation.json:411]: As mentioned in my previous comment, "Restrict model selection" would be a much better translation imo.

File: src/lib/i18n/locales/en-US/translation.json [hunk 6]
```diff
@@ -408,8 +408,8 @@
 	"Model not selected": "",
 	"Model Params": "",
 	"Model updated successfully": "",
-	"Model Whitelisting": "",
-	"Model(s) Whitelisted": "",
+	"Model Allowlisting": "",
```
Comment (kaiiiiiiiii) [inline src/lib/i18n/locales/en-US/translation.json:411]: "Restrict model selection"

File: src/lib/i18n/locales/en-GB/translation.json [hunk 7]
```diff
@@ -408,8 +408,8 @@
...
+	"Model(s) Allowlisted": "",
```
Comment (kaiiiiiiiii) [inline src/lib/i18n/locales/en-GB/translation.json:412]: "model(s) allowed" or maybe permitted. But I think allowed is just fine.

File: src/lib/i18n/locales/en-US/translation.json [hunk 8]
```diff
@@ -408,8 +408,8 @@
...
+	"Model(s) Allowlisted": "",
```
Comment (kaiiiiiiiii) [inline src/lib/i18n/locales/en-US/translation.json:412]: "model(s) allowed"

**Review by kaiiiiiiiii — COMMENTED** [review]
Hey man, I've added a couple of comments on the translations which might sound better/be more specific :)

image

**Pull request closed without merging**